File: Selenium/Activities/Activity21.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.support.wait import WebDriverWait
import selenium.webdriver.support.expected_conditions as EC


with webdriver.Firefox() as driver:
    
    explicitWait = WebDriverWait(driver,5)
    driver.get("https://training-support.net/webelements/tabs");
    print(f"The page title is : {driver.title}")
    
    parentWindow = driver.current_window_handle
    print(f"Parent Window handle is : {parentWindow}")
			
    btnNewTab =  driver.find_element(By.XPATH,"//button[text()='Open A New Tab']")
    btnNewTab.click()
			
    explicitWait.until(EC.number_of_windows_to_be(2))
    print(f"Number of tabs opened : {len(driver.window_handles)}")
			
    for handle in driver.window_handles :
        print(f"Window handle is : {handle}")
        driver.switch_to.window(handle)

    explicitWait.until(EC.element_to_be_clickable(driver.find_element(By.XPATH,"//button[text()='Open Another One']")))
    print(f"The new Tab Title : {driver.title}")
    print(f"The message is : {(driver.find_element(By.CSS_SELECTOR,"h2.mt-5")).text}")
	
    secondParentWindow = driver.current_window_handle
    print(f"Second Parent Window handle is : {secondParentWindow}")
    
    btnAnotherOne = driver.find_element(By.XPATH,"//button[text()='Open Another One']")
    btnAnotherOne.click();
    explicitWait.until(EC.number_of_windows_to_be(3));
			
    for  handle in driver.window_handles :
        print(f"Window handle is : {handle}")
        driver.switch_to.window(handle)

    explicitWait.until(EC.element_to_be_clickable(driver.find_element(By.XPATH,"//button[text()='Open Another One']")))
			
    print(f"The second new Tab Title : {driver.title}")
    print(f"The message is : {(driver.find_element(By.CSS_SELECTOR,"h2.mt-5")).text}")
			
    driver.quit()

File: Selenium/Activities/Activity21.java
package examples;

import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.WebDriverWait;

import java.time.Duration;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;

public class Activity21 {

	public static void main(String[] args) {

			WebDriver driver = new FirefoxDriver();
			WebDriverWait explicitWait = new WebDriverWait(driver,Duration.ofSeconds(5));
			
			driver.get("https://training-support.net/webelements/tabs");
			System.out.println("The page title is : "+ driver.getTitle());
			String parentWindow = driver.getWindowHandle();
			System.out.println("Parent Window handle is : "+parentWindow);
			
			WebElement btnNewTab =  driver.findElement(By.xpath("//button[text()='Open A New Tab']"));
			btnNewTab.click();
			
			explicitWait.until(ExpectedConditions.numberOfWindowsToBe(2));
			System.out.println("Number of tabs opened : " + driver.getWindowHandles().size());
			
			for(String handle : driver.getWindowHandles()) {
				System.out.println("Window handle is : "+handle);
				driver.switchTo().window(handle);
			}
			
			explicitWait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[text()='Open Another One']")));
			
			System.out.println("The new Tab Title : " + driver.getTitle());
			System.out.println("The message is : "+(driver.findElement(By.cssSelector("h2.mt-5")).getText()));
			
			String secondParentWindow = driver.getWindowHandle();
			System.out.println("Second Parent Window handle is : "+secondParentWindow);
			WebElement btnAnotherOne = driver.findElement(By.xpath("//button[text()='Open Another One']"));
			btnAnotherOne.click();
			
			explicitWait.until(ExpectedConditions.numberOfWindowsToBe(3));
			
			for(String handle : driver.getWindowHandles()) {
				System.out.println("Window handle is : "+handle);
				driver.switchTo().window(handle);
			}
			
			explicitWait.until(ExpectedConditions.elementToBeClickable(By.xpath("//button[text()='Open Another One']")));
			
			System.out.println("The second new Tab Title : " + driver.getTitle());
			System.out.println("The message in second new tab is : "+(driver.findElement(By.cssSelector("h2.mt-5")).getText()));
			
			driver.quit();

	}

}
